gst/nnstreamer/tensor_common_pipeline.c

```diff
@@ -374,20 +377,22 @@ gst_tensor_time_sync_buffer_from_collectpad (GstCollectPads * collect,
     }
 
     if (GST_IS_BUFFER (buf)) {
-      guint i, n_mem;
-
       n_mem = gst_buffer_n_memory (buf);
 
       /** These are internal logic error. If given inputs are incorrect,
           the negotiation should have been failed before this stage. */
-      if (!gst_tensors_info_is_flexible (&in_configs.info))
+      if (gst_tensors_config_is_static (&in_configs))
         g_assert (n_mem == in_configs.info.num_tensors);
       g_assert ((counting + n_mem) < NNS_TENSOR_SIZE_LIMIT);
 
+      if (gst_tensors_config_is_flexible (&in_configs))
+        configs->format = _NNS_TENSOR_FORMAT_FLEXIBLE;
+
       for (i = 0; i < n_mem; ++i) {
-        mem = gst_buffer_get_memory (buf, i);
-        gst_buffer_append_memory (tensors_buf, mem);
+        in_mem[counting] = gst_buffer_get_memory (buf, i);
+
         configs->info.info[counting] = in_configs.info.info[i];
+        in_formats[counting] = in_configs.format;
```

Comment:
Q1. Why do we need `counting`?
Q2. Is it necessary to save the format of the input for each memory?

Reply:
1. max number of tensors in gst-buffer is 16. 
2. if tensor-mux has two sink pads with different format - 1st pad is static and 2nd pad is flexible, output buffer should be flexible and tensor-mux should append header in the memories from only 1st pad. The memories from 2nd pad already have header in each mem block because it is flexble tensor.